library/src/compute/construct.rs

```diff
@@ -490,6 +499,71 @@ cast_from_value! {
     v: Str => Self(v),
 }
 
+/// Converts a character into its corresponding code point.
+///
+/// ## Example
+/// ```example
+/// #str.to-unicode("a") \
+/// #"a\u{0300}".codepoints().map(str.to-unicode)
+/// ```
+///
+/// Display: String To Unicode
+/// Category: construct
+/// Returns: int
+#[func]
+pub fn to_unicode(
+    /// The char that should be converted.
+    value: Char,
+) -> Value {
+    Value::Int(From::<u32>::from(value.0.into()))
+}
+
+struct Char(char);
+
+cast_from_value! {
+    Char,
+    v: Str => {
+        match v.chars().next() {
+            Some(c) if c.len_utf8() == v.len() as usize => Self(c),
+            _ => Err(eco_format!(
+                "string must contain exactly one code point, contained {}",
+                v.chars().count(),
+            ))?,
+        }
+    }
+}
+
+#[func]
+/// Converts a unicode code point into its corresponding string.
+///
+/// ```example
+/// #str.from-unicode(97)
+/// ```
+///
+/// Display: Sting From Unicode
+/// Category: construct
+/// Returns: string
+pub fn from_unicode(
+    /// The code point that should be converted.
+    value: CodePoint,
+) -> Value {
+    Value::Str(format_str!("{}", value.0))
+}
+
+/// The numeric representation of a single unicode code point.
+struct CodePoint(char);
+
+cast_from_value! {
+    CodePoint,
+    v: i64 => {
+        if let Some(c) = v.try_into().ok().and_then(|v: u32| v.try_into().ok()) {
+            Self(c)
+        } else {
+            Err(eco_format!("{:#x} is not inside the valid code point range", v))?
+        }
```

Comment:
i think this would be more succinctly written with `ok_or_else`

Reply:
the error message should perhaps just state "{:#x} is not a valid codepoint" since Unicode is not one consecutive range, as far as I can tell.

Reply:
Sorry, I should have made myself more clear. I made something like this:
```rust
v: i64 => {
    Self(v.try_into().ok().and_then(|v: u32| v.try_into().ok()).ok_or_else(
        || eco_format!("{:#x} is not inside the valid code point range", v),
    )?)
}
```